GG/src/ListBox.cpp

```diff
@@ -1441,10 +1460,20 @@ void ListBox::AddPaddingAtEnd(bool enable)
 { m_add_padding_at_end = enable; }
 
 void ListBox::AllowDropType(const std::string& str)
-{ m_allowed_drop_types.insert(str); }
+{
+    // Create the set if necessary
+    if (!m_allowed_drop_types)
+        m_allowed_drop_types = std::unordered_set<std::string>();
+    m_allowed_drop_types->insert(str);
+}
 
 void ListBox::DisallowDropType(const std::string& str)
-{ m_allowed_drop_types.erase(str); }
+{
+    // Create the set if necessary
+    if (!m_allowed_drop_types)
+        m_allowed_drop_types = std::unordered_set<std::string>();
```

Comment:
If `DisallowDropType` is called when m_allowed_drop_types is `none`, that will presumably disallow all drop types? A bit awkward / potentially unexpected behaviour...